src/ripple/app/tests/common_ledger.h

```diff
@@ -0,0 +1,289 @@
+    auto tx = STTx(*sopTrans);
+    if (sign)
+        tx.sign(account.sk);
+    return tx;
+}
+
+// Helper function to apply a transaction to a ledger
+template <class = void>
+void
+applyTransaction(Ledger::pointer const& ledger, STTx const& tx, bool check = true)
+{
+    TransactionEngine engine(ledger);
+    bool didApply = false;
+    auto r = engine.applyTransaction(tx, tapOPEN_LEDGER | (check ? tapNONE : tapNO_CHECK_SIGN),
+                                        didApply);
+    if (r != tesSUCCESS)
+        throw std::runtime_error(
+            "r != tesSUCCESS");
+    if (! didApply)
+        throw std::runtime_error(
+            "didApply");
+}
+
+// Create genesis ledger from a start amount in drops, and the public
+// master RippleAddress
+template <class = void>
+Ledger::pointer
+createGenesisLedger(std::uint64_t start_amount_drops, TestAccount const& master)
+{
+    Ledger::pointer ledger = std::make_shared<Ledger>(master.pk,
+                                                        start_amount_drops);
+    ledger->updateHash();
+    ledger->setClosed();
+    if (! ledger->assertSane())
+        throw std::runtime_error(
+        "! ledger->assertSane()");
+    return ledger;
+}
+
+// Create an account represented by public RippleAddress and private
+// RippleAddress
+template <class = void>
+TestAccount
+createAccount(std::string const& passphrase)
+{
+    RippleAddress const seed
+            = RippleAddress::createSeedGeneric (passphrase);
+    RippleAddress const generator
+            = RippleAddress::createGeneratorPublic (seed);
+
+    return {
+        std::move (RippleAddress::createAccountPublic (generator, 0)),
+        std::move (RippleAddress::createAccountPrivate (generator, seed, 0)),
+        std::uint64_t (0)
+    };
+}
+
+template <class = void>
+void
+freezeAccount(TestAccount& account, Ledger::pointer const& ledger, bool sign = true)
+{
+    Json::Value tx_json;
+    tx_json["TransactionType"] = "AccountSet";
+    tx_json["Fee"] = std::to_string(10);
+    tx_json["Account"] = account.pk.humanAccountID();
+    tx_json["SetFlag"] = asfGlobalFreeze;
+    tx_json["Sequence"] = ++account.sequence;
+    STTx tx = parseTransaction(account, tx_json, sign);
+    applyTransaction(ledger, tx, sign);
+}
+
+template <class = void>
+void
+unfreezeAccount (TestAccount& account, Ledger::pointer const& ledger, bool sign = true)
+{
+    Json::Value tx_json;
+    tx_json["TransactionType"] = "AccountSet";
+    tx_json["Fee"] = std::to_string(10);
+    tx_json["Account"] = account.pk.humanAccountID();
+    tx_json["ClearFlag"] = asfGlobalFreeze;
+    tx_json["Sequence"] = ++account.sequence;
+    STTx tx = parseTransaction(account, tx_json, sign);
+    applyTransaction(ledger, tx, sign);
+}
+
+template <class = void>
+STTx
+makePayment(TestAccount& from, TestAccount const& to,
+            std::uint64_t amountDrops,
+            Ledger::pointer const& ledger, bool sign = true, bool apply = true)
```

Comment:
It's also worth thinking about how far the `bool` must travel.  One piece of rippled code I was maintaining had a `bool` that was passed through at least 4 layers of calls.  That `bool` was sometimes joined by other `bool`s along the way.  Once I understood the situation, as a defensive measure, I changed that `bool` to an `enum`.  I'm only too aware of how capable I am of making mistakes.  By converting the `bool` to an `enum` I got guidance from the compiler that I was tracking the right parameter.  To me it seemed (and still seems) a win, despite the fact that it uses more characters.  `bool`s do have their place.  They are great as return values for testing.